controllers/upgrade_handlers.go

```diff
@@ -319,10 +333,12 @@ func (u *UpgHandler) PostPivot(ctx context.Context, ibu *lcav1alpha1.ImageBasedU
 			u.autoRollbackIfEnabled(ibu, fmt.Sprintf("Rollback due to restore failure: %s", err))
 			return doNotRequeue(), nil
 		}
+		utils.SetUpgradeStatusInProgress(ibu, fmt.Sprintf("Restoring OADP Data: Failure occurred: %s", err))
```

Comment:
glad we are making use of the status msg more. but i bet we (i) will forget to update the status right before returning with an error. Is there something we can do like in prep where updating the msg is sort of handled implicitly? (dont have to update this now...just a thought)

Reply:
The Prep handler status updates need work, as well, as they generally lag behind what's actually happening, sometimes quite a bit, but at least part of that would be due to the structure of the Prep handler where tasks are run in the background. That's not the case in the Upgrade handler, where the status can be updated directly as needed.